starport-blog/01-index.md

```diff
@@ -0,0 +1,345 @@
+
+Posts in our key-value store will look like this:
+
+```
+"post-0bae9f7d-20f8-4b51-9d5c-af9103177d66": {
+  "Creator": "cosmos18cd5t4msvp2lpuvh99rwglrmjrrw9qx5h3f3gz",
+  "Title": "This is a post!",
+  "ID": "0bae9f7d-20f8-4b51-9d5c-af9103177d66"
+},
+"post-8c6d8cd4-b4c9-4ba3-a683-e894db3f2605": {
+  ...
+}
+```
+
+Right now the store is empty. Let's figure out how to add posts.
+
+With the Cosmos SDK, users can interact with your app with either a CLI (`blogcli`) or by sending HTTP requests. Let's define the CLI command first. Users should be able to type `blogcli tx blog create-post "This is a post!" --from=user1` to add a post to your store. The `create-post` subcommand hasn’t been defined yet--let’s do it now.
+
+## x/blog/client/cli/tx.go
+
+In the `import` block, make sure to import these six additional packages:
+
+```go
+import (
+  // Existing imports...
+  "bufio"
+  "github.com/cosmos/cosmos-sdk/client/context"
+  "github.com/cosmos/cosmos-sdk/x/auth"
+  "github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+  sdk "github.com/cosmos/cosmos-sdk/types"
+)
+```
+
+This file already contains `func GetTxCmd` which defines custom `blogcli` commands. We will add the custom `create-post` command to our `blogcli` by first adding `GetCmdCreatePost` to `blogTxCmd`.
+
+```go
+  blogTxCmd.AddCommand(flags.PostCommands(
+    GetCmdCreatePost(cdc),
+  )...)
+```
+
+At the end of the file, let's define `GetCmdCreatePost` itself.
+
+```go
+func GetCmdCreatePost(cdc *codec.Codec) *cobra.Command {
+  return &cobra.Command{
+    Use:   "create-post [title]",
+    Short: "Creates a new post",
+    Args:  cobra.ExactArgs(1),
+    RunE: func(cmd *cobra.Command, args []string) error {
+      cliCtx := context.NewCLIContext().WithCodec(cdc)
+      inBuf := bufio.NewReader(cmd.InOrStdin())
+      txBldr := auth.NewTxBuilderFromCLI(inBuf).WithTxEncoder(utils.GetTxEncoder(cdc))
+      msg := types.NewMsgCreatePost(cliCtx.GetFromAddress(), args[0])
+      err := msg.ValidateBasic()
+      if err != nil {
+        return err
+      }
+      return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+    },
+  }
+}
+```
+
+The function above defines what happens when you run the `create-post` subcommand. `create-post` takes one argument `[title]`, creates a message `NewMsgCreatePost` (with title as `args[0]`) and broadcasts this message to be processed in your application.
+This is a common pattern in the SDK: users make changes to the store by broadcasting messages. Both CLI commands and HTTP requests create messages that can be broadcasted in order for state transition to occur.
+
+Let’s define `NewMsgCreatePost` in a new file you should create as `x/blog/types/MsgCreatePost.go`.
+
+## x/blog/types/MsgCreatePost.go
+
```

Comment:
It's a bit weird to start a section with a code block. A short intro sentence would be nice

Reply:
maybe just take the previous sentence and put it there